File: python3/step0_repl.py
import readline  # so input() uses editable input
# import sys


def READ(line):
    return line


def EVAL(line):
    return line


def PRINT(line):
    return line


def rep(line):
    return PRINT(EVAL(READ(line)))


def main():
    print("MAL in Python3")
    print()

    while True:
        try:
            line = input("user> ")
        except EOFError:
            print("Bye")
            # sys.exit(0)
            break
        result = rep(line)
        print(result)

if __name__ == '__main__':
    main()

File: java/src/main/java/mal/step0_repl.java
package mal;

import java.io.Console;

public class step0_repl {
  public static void main(String args[]) {
    Console console = System.console();
    String input;
    String output;
    
    while (true) {
      input = console.readLine("user> ");
      if (input == null) {      // Test for EOF
        break; 
      }
      else {
        output = rep(input);
      }
      System.out.println(output);
    }
  }

  public static String READ(String arg) {
    return arg;
  }

  public static String EVAL(String arg) {
    return arg;
  }

  public static String PRINT(String arg) {
    return arg;
  }

  public static String rep(String arg) {
    String result;

    result = PRINT(EVAL(READ(arg)));
    return result;
  }
}